src/lib/handler-filter.js

```diff
@@ -27,17 +27,23 @@ function addRequestHeaderCount(handler) {
 
 exports.filterHandlers = function (req, handlers, ignoreHeaders) {
     if (handlers) {
-        var filteredHandlers;
+        let filteredHandlers;
 
-        handlers.sort(content.contentTypeComparator);
+        filteredHandlers = handlers.filter(filterRequestHeader(req, ignoreHeaders));
+        if (filteredHandlers.length === 0) {
+            return route.createErrorHandler(handlers[0],["No handler matches after filtering required headers (including content-type)"])
```

Comment:
The error message explains what happened inside Drakov but doesn't tell the end-user what is wrong with their request or what they should do different. Could this be expressed as `Found URL but request doesn't contain required headers to match against any known fixture` and the second error as `Found URL but request doesn't contain required query parameters to match against any known fixture`